www/include/views/graphs/exportData/ExportCSVServiceData.php

```diff
@@ -58,58 +58,99 @@ function get_error($str)
+                $index = $row['id'];
+            }
+        }
     } else {
         die('Resource not found');
     }
 }
+if ($index !== false) {
+    $stmt = $pearDBO->prepare(
+        'SELECT host_name, service_description FROM index_data WHERE id = :index'
+    );
+    $stmt->bindValue(':index', $index, \PDO::PARAM_INT);
+    $stmt->execute();
+    while ($res = $stmt->fetchRow()) {
+        $hName = $res["host_name"];
+        $sName = $res["service_description"];
+    }
 
-$period = (isset($_POST["period"])) ? htmlentities($_POST["period"], ENT_QUOTES, "UTF-8") : "today";
-$period = (isset($_GET["period"])) ? htmlentities($_GET["period"], ENT_QUOTES, "UTF-8") : $period;
+    header("Content-Type: application/csv-tab-delimited-table");
+    if (isset($hName) && isset($sName)) {
+        header("Content-disposition: filename=" . $hName . "_" . $sName . ".csv");
+    } else {
+        header("Content-disposition: filename=" . $index . ".csv");
+    }
 
-$DBRESULT = $pearDBO->query("SELECT host_name, service_description FROM index_data WHERE id = '$index'");
-while ($res = $DBRESULT->fetchRow()) {
-    $hName = $res["host_name"];
-    $sName = $res["service_description"];
-}
+    $listMetric = array();
+    $datas = array();
+    $listEmptyMetric = array();
 
-header("Content-Type: application/csv-tab-delimited-table");
-if (isset($hName) && isset($sName)) {
-    header("Content-disposition: filename=" . $hName . "_" . $sName . ".csv");
-} else {
-    header("Content-disposition: filename=" . $index . ".csv");
-}
+    $stmt = $pearDBO->prepare(
+        'SELECT DISTINCT metric_id, metric_name ' .
+        'FROM metrics, index_data ' .
+        'WHERE metrics.index_id = index_data.id AND id = :index ORDER BY metric_name'
+    );
+
+    $stmt->bindValue(':index', $index, \PDO::PARAM_INT);
+    $stmt->execute();
 
-$listMetric = array();
-$datas = array();
-$listEmptyMetric = array();
-
-$query = "SELECT DISTINCT metric_id, metric_name FROM metrics, index_data " .
-    "WHERE metrics.index_id = index_data.id AND id = '$index' ORDER BY metric_name";
-$DBRESULT = $pearDBO->query($query);
-while ($index_data = $DBRESULT->fetchRow()) {
-    $listMetric[$index_data["metric_id"]] = $index_data["metric_name"];
-    $listEmptyMetric[$index_data["metric_id"]] = '';
-    $query2 = "SELECT ctime,value FROM data_bin WHERE id_metric = '" . $index_data["metric_id"] .
-        "' AND ctime >= '" . htmlentities($_GET["start"], ENT_QUOTES, "UTF-8") .
-        "' AND ctime < '" . htmlentities($_GET["end"], ENT_QUOTES, "UTF-8") . "'";
-    $DBRESULT2 = $pearDBO->query($query2);
-    while ($data = $DBRESULT2->fetchRow()) {
-        $datas[$data["ctime"]][$index_data["metric_id"]] = $data["value"];
+    while ($indexData = $stmt->fetch(\PDO::FETCH_ASSOC)) {
+        $listMetric[$indexData['metric_id']] = $indexData['metric_name'];
+        $listEmptyMetric[$indexData['metric_id']] = '';
+        if (isset($start) && isset($end)) {
```

Comment:
```suggestion
        if ($star !== false && $end !== false) {
```

Reply:
what should happen if $start or $end is false ? an error displayed or something thrown ?

Reply:
It should not happen since the caller defines default date times for start and end

Reply:
Agreed with sc979, as we initially define in the filter_var() that $start and $end could be false.